File: 2215-find-the-difference-of-two-arrays/2215-find-the-difference-of-two-arrays.cpp
class Solution {
public:
    vector<vector<int>> findDifference(vector<int>& nums1, vector<int>& nums2) {
        vector<vector<int>> result;
        vector<int> v1, v2;
        set<int> n1 (nums1.begin(), nums1.end()), n2(nums2.begin(), nums2.end());
        
        for(int i:n1)
        {
            if(n2.find(i)==n2.end())
            {
                v1.push_back(i);
            }
        }
        for(int i:n2)
        {
            if(n1.find(i)==n1.end())
            {
                v2.push_back(i);
            }
        }
        result.push_back(v1);
        result.push_back(v2);
        return result;
    }
};

File: 2215-find-the-difference-of-two-arrays/2215-find-the-difference-of-two-arrays.java
class Solution {
    public List<List<Integer>> findDifference(int[] nums1, int[] nums2) {
        Set<Integer> set1 = new HashSet<>();
        Set<Integer> set2 = new HashSet<>();
        
        for(int i: nums1)
        {
            set1.add(i);
        }
        for(int i: nums2)
        {
            set2.add(i);
            set1.remove(i);
        }
        for(int i:nums1)
        {
            set2.remove(i);
        }
        List<List<Integer>> result=new ArrayList<>();
        result.add(new ArrayList<>(set1));
        result.add(new ArrayList<>(set2));
        return result;
    }
}